easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1668,6 +1668,7 @@ def det_short_module_name(self, ec, force_visible=False):
         self.log.debug("Obtained valid short module name %s" % mod_name)
 
         # sanity check: obtained module name should pass the 'is_short_modname_for' check
+        ec = self.check_ec_type(ec)
```

Comment:
@bartoldeman You should include this at the top of `det_short_module_name`, before you pass down `ec`, since otherwise you may be tranforming `ec` into a full `EasyConfig` instance twice in the same `det_short_module_name` call